crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -218,13 +229,29 @@ impl Inner {
         );
 
         // surplus and fees calculation
-        let surplus = settlement.total_surplus(&external_prices);
+        let competition_order_uids = Postgres::find_competition(auction_id, ex)
+            .await?
+            .context(format!(
+                "missing competition for auction_id={:?}",
+                auction_id
+            ))?
+            .common
+            .auction
+            .orders
+            .into_iter()
+            .map(|order| OrderUid(order.0))
+            .collect::<HashSet<_>>();
+        let surplus_exempt_order_uids = Postgres::get_missing_order_uids(
+            settlement.trades.iter().filter_map(|trade| {
+                // check persistence of the orders not in the competition
+                (!competition_order_uids.contains(&trade.order_uid)).then_some(trade.order_uid)
+            }),
+            ex,
+        )
+        .await?;
```

Comment:
Why do we still need to do this? We now have access to the auction, so anything that is in the settlement and not in the auction is exempt from surplus. We should not need this extra db roundtrip (this query is my main worry with the original PR)

Reply:
CoW AMM JIT orders will not be in the auction or DB but are still supposed to receive surplus.